### ISSUE TITLE
[Request] dcp/dsync: add support to remote src&dest

### ISSUE THREAD

**Issue by KnightKu** [hpc/mpifileutils#303]
dcp/dysnc only works when clients mount both filesystem (src/dst), if people want dcp/dsync over the network, but no mount points. So we might need to have something like "dcp -e ssh user@ip:/src /dest" that rsync is doing, means we need add support to ssh stream into libmfu (MPI aware ssh I/O?).
With a pre-investigation, seems it's not easy to do so. And @adammoody @dinatale2 , can you give some advices here? thanks

**Comment by adammoody** [comment]
@KnightKu , that's right.  MPI and ssh don't generally work together this way.  It's not that it's prohibited in the MPI standard, but it's just not a commonly requested use case, and to my knowledge it is not well supported across MPI libraries.

Perhaps there are some grid-based MPI libraries for that, like https://www.globus.org/sites/default/files/gempi.pdf?

To do this effectively, one would need to run a group of MPI procs on the local system, and another group of MPI procs on the remote system, and then somehow connect the two.  So there is one challenge in getting both groups launched on the two systems at the same time.  There is a second challenge in getting them connected and securely exchanging data.

**Comment by KnightKu** [comment]
@adammoody yeah, indeed, we might need to manage the processes and the communication our selves.
Thanks for you comments, will have a look into this paper.
